### PR TITLE
PTV-1786: Extract type methods from the server class into their own class

### PR DESCRIPTION
**Pull request by MrCreosote**
Will allow creating an interface and alternate implementation that delegates to another workspace service

### CODE REVIEW

File: src/us/kbase/workspace/kbase/LocalTypeServerMethods.java [hunk 1]
```diff
@@ -0,0 +1,323 @@
+package us.kbase.workspace.kbase;
```
Comment (MrCreosote) [inline src/us/kbase/workspace/kbase/LocalTypeServerMethods.java:1]: This PR is effectively just moving all the code from `WorkspaceServer` that deals with the type database in here, almost verbatim, plus the two methods from `WorkspaceServerMethods`.

File: src/us/kbase/workspace/kbase/LocalTypeServerMethods.java [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+		}
+		final List<String> add = params.getNewTypes() != null ?
+				params.getNewTypes() : new ArrayList<String>();
+		final List<String> rem = params.getRemoveTypes() != null ?
+				params.getRemoveTypes() : new ArrayList<String>();
+		final Map<String, Long> deps = params.getDependencies() != null ?
+				params.getDependencies() : new HashMap<String, Long>();
+		final Map<TypeDefName, TypeChange> res;
+		if (params.getMod() != null) {
+			 res = types.compileTypeSpec(
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/kbase/LocalTypeServerMethods.java:112]: is this going to be covered somewhere else / in a future PR?

Comment (MrCreosote) [reply src/us/kbase/workspace/kbase/LocalTypeServerMethods.java:112]: Not in the near future anyway. It hasn't been covered for the last 8 years apparently, so for now I'm leaving things as is.

**Review by ialarmedalien — APPROVED** [review]
Looks fine

**Pull request merged**